components/CheckBox/CheckBox.tsx

```diff
@@ -65,15 +71,45 @@ function CheckBox({ filters, handleFilters }) {
     } else {
       newCheck.splice(currentId, 1);
     }
+
     setChecked(newCheck);
     localStorage.setItem("checked", JSON.stringify(newCheck));
     handleFilters(newCheck);
   };
 
-  function isChecked(id) {
-    return !(Checked.indexOf(id) === -1);
+
+  const allChecked = (index: number) => {
+    return !(event[index].some((elem) => { return !Checked.includes(elem.id) }))
+
+  }
+  const noneChecked = (index: number) => {
+    return !event[index].some((elem) => { return Checked.includes(elem.id) })
   }
 
+  const handleToggleAll = (values, index: number, index1: number) => {
+    const newCheck = [...Checked];
+
+    if (!allChecked(index1 + sync[index]))
+      for (const value of values) {
+        if (!newCheck.includes(value.id)) {
+          newCheck.push(value.id);
+        }
+        setChecked(newCheck);
+        localStorage.setItem("checked", JSON.stringify(newCheck));
+        handleFilters(newCheck);
+      }
+    else
+      for (const value of values) {
+        if (newCheck.includes(value.id)) {
+          newCheck.splice(newCheck.indexOf(value.id));
+        }
+        setChecked(newCheck);
+        localStorage.setItem("checked", JSON.stringify(newCheck));
+        handleFilters(newCheck);
+      }
+  };
```

Comment:
also, I'd suggest a easier way by doing this function, such as:

```js
  const handleToggleAll = (values, index: number, index1: number) => {
    let newChecked = [...Checked];

    // If all checkboxed are not checked, check them all.
    // Else, filter the checked events that was not finded in `values`
    if (!isAllChecked(index1 + sync[index])) {
      values.map((event) => newChecked.push(event.id));
    } else {
      newChecked = newChecked.filter(
        (eventId) => !values.find((value) => value.id === eventId)
      );
    }

    setChecked(newChecked);
    handleFilters(newChecked);
    localStorage.setItem("checked", JSON.stringify(newChecked));
  };
```

I think it is a more concise and maintainable way of writing this function.

Reply:
Thank you!